.github/workflows/equivalence-test.yml

```diff
@@ -1,27 +1,238 @@
 name: Terraform Equivalence Tests
 
-# This action will execute the suite of Terraform equivalence tests after a
-# tag has been pushed for a new version.
-#
-# For now, it is just a skeleton action that will be populated shortly.
-
 on:
   workflow_dispatch:
     inputs:
-      version:
+      terraform-version:
```

Comment:
There is a way to do this, but it's not straight forward. I'll pull together a follow up PR that introduces this and share it with you.

Reply:
I have an additional idea today which didn't occur to me last week:

The build workflow produces a `metadata.json` artifact which I believe is supposed to support a very similar idea of "accept just workflow run ID, then programmatically determine relevant metadata" for the main release process. In the most recent build at the time I'm writing this, it looks like this:

```hcl
{
		"branch": "main",
		"buildworkflowid": "3220439158",
		"product": "terraform",
		"repo": "terraform",
		"org": "hashicorp",
		"sha": "bc1436af530f41af048ea5fc17ae87759e1c483e",
		"version": "1.3.0-alpha20220817-187-gbc1436af53"
}
```

As you can see, this includes a `version` property which includes the version number that the build workflow decided. Since this particular artifact has a reliable static name (which doesn't vary between runs), it's hopefully easier to fetch this one first and then derive the expected filename for the distribution `.zip` file from it, rather than either trying to consume the annotation (as I previously suggested) or by trying to fuzzily-match the artifact filenames from the GitHub artifacts endpoint.

I'm not sure if that ends up being easier than the way you had in mind (or maybe this _is_ the way you had in mind!), but I believe it is at least already implemented this way in the internal release pipeline so maybe there's some code there we can borrow from, assuming it doesn't depend on anything sensitive.

Reply:
Okay, I've checked this out finally. And the `metadata.json` approach will work and is the simplest approach.

I will merge this as is, however, and pull together another PR to make this change.